File: app/src/main/java/com/example/aol_mcs/MainActivity.java
package com.example.aol_mcs;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.android.volley.Request;
import com.android.volley.Response;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView dollRecyclerView;
    private DollAdapter dollAdapter;
    private final ArrayList<Doll> dollList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.home_page);

        // Initialize UI components
        dollRecyclerView = findViewById(R.id.dollRecyclerView);
        dollRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        dollAdapter = new DollAdapter(dollList, selectedDoll -> {
            // Handle doll item click, e.g., redirect to doll detail page
            redirectToDollDetail(selectedDoll);
            return null;
        });
        dollRecyclerView.setAdapter(dollAdapter);

        // Fetch doll data from JSON using Volley
        fetchDollDataFromJson();
    }

    private void fetchDollDataFromJson() {
        String url = "https://api.npoint.io/9d7f4f02be5d5631a664";

        JsonObjectRequest request = new JsonObjectRequest(Request.Method.GET, url, null,
                response -> parseJsonResponse(response),
                error -> showToast("Error fetching doll data: " + error.getMessage()));

        // Add the request to the RequestQueue.
        Volley.newRequestQueue(this).add(request);
    }

    private void parseJsonResponse(JSONObject response) {
        try {
            JSONArray jsonArray = response.getJSONArray("dolls");

            for (int i = 0; i < jsonArray.length(); i++) {
                JSONObject dollObject = jsonArray.getJSONObject(i);

                // Extract relevant doll data from JSON
                int dollId = dollObject.getInt("id");
                String dollName = dollObject.getString("name");
                String dollCover = dollObject.getString("cover");

                // Create Doll object and add to the list
                Doll doll = new Doll(dollId, dollName, dollCover);
                dollList.add(doll);
            }

            // Update RecyclerView with doll data
            dollAdapter.notifyDataSetChanged();

        } catch (JSONException e) {
            showToast("Error parsing JSON: " + e.getMessage());
        }
    }

    private void showToast(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    private void redirectToDollDetail(Doll selectedDoll) {
        // Implement redirection to Doll Detail Page
        // You can use Intent to start a new activity and pass relevant data
        // For example:
        Intent intent = new Intent(this, Doll.class);
        intent.putExtra("dollId", selectedDoll.getId());
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/aol_mcs/MainActivity.kt
package com.example.aol_mcs

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.aol_mcs.R
import kotlinx.android.synthetic.main.home_page.*
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var dollAdapter: DollAdapter
    private val dollList = mutableListOf<Doll>()

    // Declare a variable to keep track of the selected menu item
    private var selectedMenuItemId: Int = R.id.menu_home

    private val onNavigationItemSelectedListener =
        BottomNavigationView.OnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.menu_home -> {
                    if (selectedMenuItemId != R.id.menu_home) {
                        // Clear any existing views from FrameLayout
                        fragment_container.removeAllViews()
                        // Notify adapter about the change
                        dollAdapter.notifyDataSetChanged()
                        // Fetch doll data and populate the RecyclerView
                        getDollData()
                        selectedMenuItemId = R.id.menu_home
                    }
                    true
                }
                R.id.menu_transactions -> {
                    if (selectedMenuItemId != R.id.menu_transactions) {
                        // Clear the RecyclerView when navigating to the transactions page
                        dollList.clear()
                        dollAdapter.notifyDataSetChanged()
                        startActivity(Intent(this, TransactionActivity::class.java))
                        selectedMenuItemId = R.id.menu_transactions
                    }
                    true
                }
                R.id.menu_profile -> {
                    if (selectedMenuItemId != R.id.menu_profile) {
                        // Clear the RecyclerView when navigating to the profile page
                        dollList.clear()
                        dollAdapter.notifyDataSetChanged()
                        // Inflate profile page layout
                        val profileLayout = layoutInflater.inflate(R.layout.profile_page, null)
                        fragment_container.addView(profileLayout)
                        selectedMenuItemId = R.id.menu_profile
                    }
                    true
                }
                else -> {
                    // Update selectedMenuItemId for other menu items
                    selectedMenuItemId = item.itemId
                    // Return true to indicate that the item is handled
                    true
                }
            }
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.home_page)

        val navView: BottomNavigationView = findViewById(R.id.bottom_navigation)
        navView.setOnNavigationItemSelectedListener(onNavigationItemSelectedListener)

        recyclerView = findViewById(R.id.recycler_view)
        dollAdapter = DollAdapter(this, dollList,
            onItemClick = { doll ->
                val intent = Intent(this, DollDetailActivity::class.java)
                intent.putExtra("doll", doll)
                startActivity(intent)
            },
            onBuyClick = { doll ->
                // Handle buy click action here if needed
            }
        )

        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = dollAdapter

        getDollData()
    }

    private fun getDollData() {
        // Fetch doll data from your database or any other source
        // For demo, I'm adding dummy data
        dollList.clear() // Clear existing data
        dollList.addAll(getDummyDollData())
        dollAdapter.notifyDataSetChanged() // Notify adapter about the change
    }

    // Dummy function to generate doll data
    private fun getDummyDollData(): List<Doll> {
        return listOf(
            Doll(
                "SNRLX",
                "Snorlax",
                "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSGpwr2HfN4VUlGjJZbQDI_S_6LpNI9Ndkz1A&usqp=CAU",
                "Large", 4.5f, 29.99, "Description 1"
            ),
            Doll(
                "PTRCK",
                "Patrick",
                "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRriee-mWuEk9q2h0c5mYzOFjQ9yxnh6rcFDg&usqp=CAU",
                "Medium", 4.2f, 19.99, "Description 2"
            ),
            Doll(
                "SPNGB",
                "Spongebob",
                "https://media.karousell.com/media/photos/products/2023/10/23/boneka_spongebob_squarepants___1698019485_6bd22511_progressive.jpg",
                "Small", 4.0f, 9.99, "Description 3"
            )
        )
    }
}






